react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+class ToggleSwitch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isOn: true };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    if (this.state.isOn) {
+      this.setState({ isOn: false });
+    } else {
+      this.setState({ isOn: true });
+    }
+  }
+
+  render() {
+    let className = 'toggle off';
+    let spanText;
+    let circleName;
+    if (this.state.isOn) {
```

Comment:
The only update I would do here is move the logic into a different method and have that method return an object with the className, circleName, and spanText values so you can use that object to set the values in the JSX.